cxx11.cmake

```diff
@@ -0,0 +1,47 @@
+# Copyright (C) 2018 LAAS-CNRS, JRL AIST-CNRS, INRIA
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+include(CheckCXXCompilerFlag)
+
+#.rst:
+# .. ifmode:: user
+#
+# .. command:: PROJECT_USE_CXX11
+#
+#    This macro set up the project to compile the whole project 
+#    with C++11 standards.  
+#
+MACRO(PROJECT_USE_CXX11)
+  CHECK_CXX_COMPILER_FLAG("-std=c++0x" COMPILER_SUPPORTS_CXX0X)
+  CHECK_CXX_COMPILER_FLAG("-std=c++11" COMPILER_SUPPORTS_CXX11)
+  IF(COMPILER_SUPPORTS_CXX0X OR COMPILER_SUPPORTS_CXX11)
+    IF(CMAKE_VERSION VERSION_LESS "3.1")
+      IF(COMPILER_SUPPORTS_CXX0X)
+        SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x")
+      ELSE()  
+        SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+      ENDIF()
+    ELSE()
+      SET(CMAKE_CXX_STANDARD 11)
+    ENDIF()
+    IF(APPLE)
+      IF(POLICY CMP0025)
+        CMAKE_POLICY(SET CMP0025 NEW)
```

Comment:
Is this related to C++11 support? Shouldn't that be part of `APPLY_DEFAULT_APPLE_CONFIGURATION()`?

Reply:
This is very related to C++11 support. So, I think this is the right place.

Reply:
Ok, I don't mind keeping it in. Just for future reference, can you explain how this is related to C++11 support?  From my understanding, [CMP0025](https://cmake.org/cmake/help/v3.0/policy/CMP0025.html) is about the identifier of clang inside cmake scripts

Reply:
It was for suppressing some boring messages I got when CMAKE_CXX_STANDARD was used. I can remove if it only concerns my implementation.